runtime/plaid-stl/src/github/mod.rs

```diff
@@ -1023,6 +1023,49 @@ pub fn get_branch_protection_rules(
     Ok(String::from_utf8(return_buffer).unwrap())
 }
 
+/// Get protection rules (as in ruleset) for a branch
+/// ## Arguments
+///
+/// * `owner` - The account owner of the repository. The name is not case sensitive.
+/// * `repo` - The name of the repository without the .git extension. The name is not case sensitive.
+/// * `branch` - The name of the branch. Cannot contain wildcard characters.
+pub fn get_branch_protection_ruleset(
```

Comment:
I bet we could make this a macro and remove a whole bunch of code across this module